packages/run-protocol/test/test-gov-collateral.js

```diff
@@ -182,29 +167,73 @@ const makeScenario = async t => {
     return purses;
   };
 
-  const startRunPreview = async () => {
-    const {
-      installation: { produce: iProduce },
-    } = space;
-    iProduce.VaultFactory.resolve(t.context.installation.VaultFactory);
-    iProduce.liquidate.resolve(t.context.installation.liquidate);
-    iProduce.amm.resolve(t.context.installation.amm);
-    iProduce.reserve.resolve(t.context.installation.reserve);
+  /** @type {any} */
+  const { cachedInstall: produceCachedInstall } = space.produce;
+  produceCachedInstall.resolve(t.context.cachedInstall);
+  const makeEnactCoreProposalsFromSource =
+    ({ makeCoreProposalArgs, E: cpE }) =>
+    allPowers => {
+      const {
+        consume: { cachedInstall },
+      } = allPowers;
+      const restoreRef = async ref => {
+        const { source, bundle } = ref;
+        return cpE(cachedInstall)(source, bundle);
+      };
+
+      return Promise.all(
+        makeCoreProposalArgs.map(async ({ ref, call }) => {
+          const subBehavior = makeCoreProposalBehavior({
+            manifestInstallRef: ref,
+            getManifestCall: call,
+            E: cpE,
+            restoreRef,
+          });
+          await subBehavior(allPowers);
+        }),
+      );
+    };
 
-    const USD = makeIssuerKit('USD');
-    const ATOM = makeIssuerKit('ATOM');
-    space.oracleBrand.produce.USD.resolve(USD.brand);
+  /**
+   * @param {string[]} coreProposals
+   */
+  const evalProposals = async coreProposals => {
+    const { code } = await extractCoreProposalBundles(
```

Comment:
`extractCoreProposalBundles` closes over ambient `fs` and such in a way that's making this hard for me to follow.
Let's please
  1. make those overrideable defaults a la `writeFile = fs.promises.writeFile` in `makeWriteCoreProposal` and
  2. override them when we use them here in order to be explicit about authority

Reply:
Agreed.

Reply:
perhaps in a follow-on PR